### ISSUE TITLE
Why enable retain_graph in backward?

### ISSUE THREAD

**Issue by huangbiubiu** [shenweichen/DeepCTR-Torch#69]
**Describe the question(问题描述)**
Why use `retain_graph` in model backward? Does the model need backward twice?

https://github.com/shenweichen/DeepCTR-Torch/blob/caa12dd15144900ecfcb04908e9834669cb12304/deepctr_torch/models/basemodel.py#L226

**Comment by jhhugo** [comment]
because ` 222                       total_loss = loss + self.reg_loss`, self.reg_loss use the graph weight, so you should save the weight to backward

**Comment by zanshuxun** [comment]
`retain_graph=True` can be removed since v0.2.2, we have upgraded the computation of reg_loss. please use `pip install -U deepctr-torch` to upgrade.

**Comment by zanshuxun** [comment]
https://stackoverflow.com/questions/46774641/what-does-the-parameter-retain-graph-mean-in-the-variables-backward-method

**Comment by zanshuxun** [comment]
removed

**Comment by zanshuxun** [comment]
we have solved this issue in [v0.2.4](https://github.com/shenweichen/DeepCTR-Torch/releases/tag/v0.2.4), please use `pip install -U deepctr-torch` to upgrade.
